app.py

```diff
@@ -5,13 +5,15 @@
 from src.utils.requests_retry_client import RequestsRetryClient
 
 from flask import Flask, request, Response, abort, jsonify, render_template
+from flask_cors import cross_origin
 from typing import Union
 
 app = Flask(__name__)
 # Run this by poetry run flask run
 
 
 @app.route("/random_game")
+@cross_origin(origins="http://localhost:3000")
```

Comment:
How will this work in production? The domain for the production site isn't `localhost` so this probably won't work to allows Cross-Origin requests in production. Have you considered setting up a config file or something that can read the proper allowed origin for the relevant environment?

Reply:
Hmmm, we were trial and erroring with [documentation](https://flask-cors.corydolphin.com/en/latest/api.html#decorator) to get the React application to talk to the Flask endpoint without getting a CORS Same Origin error and this finally worked.  But I guess we didn't think about it running in production.  It wouldn't just be 3000 port on the digital ocean just how Flask just works on it.

Reply:
Yeah I think the better solution would be to setup the webpack proxy to let it think it's running on the same port as the BE even though it will be running on port 3000.

Reply:
@JDGiardino it's BOTH the port and the hostname, an origin consists of a 3-tuple of `(scheme, host, port)`[1] which is validated in the user-agent by comparing its origin against the `Access-Control-Allowed-Origin` header in the response. I _think_ the origin of the production server would be `https://<whatever-the-url-is-for-the-applet>:80` but you might want to double check that.

@halo123u not sure I follow, granted I'm not 100% sure how webpack works so the proxy thing is interesting. I think ultimately its the user-agent (ie the browser) that is doing the CORS check so thats the place that you need to ensure is checking the proper header value for the current origin.

The other thing I was thinking of is instead of providing a single origin in the `cross_origin` header you can perhaps specify a list of allowed origins for the backend. In this list you can specify both the origin of your local development frontend (`http://localhost:3000`) and the production frontend (which I'm still fuzzy about, see the first point for how to find this one.) This _should_ set the `Access-Control-Allowed-Origin` header properly given a request from either origin.

[1] https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

Reply:
@nickdibari I am no expert on webpack either haha but I vaguely remember webpack having some configuration for setting up [proxy](https://webpack.js.org/configuration/dev-server/#devserverproxy). I've done that in a the past a long time ago but not sure if it will work. I was think if I can get webpack to run the proxy on local development, then when the build is made, we can dump all of the js, css, and html files in the python server so that there are no CORS issues since the server is also serving the front end. I think that way we can get rid of Flask-CORS and close this pr. Will do more research and follow up with my findings.

Reply:
Ahhhh okay a list of multiple origins both one for dev and one for prod like you said Nicholas also makes sense I think.